### PR TITLE
fix: use hash value for file name

### PR DESCRIPTION
**Pull request by Robot-Inventor**
This pull request changes the filename of the cached images to use a hash value instead of a URL.

The URLs of open graph images on some sites are very long due to parameters. Caching such images may result in an error because the file name is too long.

By using hash values, the length of filenames is kept constant and duplicate images are eliminated.

### CODE REVIEW

**Comment by gladevise** [comment]
@Robot-Inventor Thank you for your pull request. I appreciate the effort you've put into improving our caching system for open graph images. Your proposed change to use hash values instead of URLs for cached image filenames seems like a sensible approach to address the issue of overly long filenames caused by lengthy URLs with parameters.

I will review your code changes thoroughly and provide feedback or suggestions if needed.

File: index.js
Comment (gladevise) [inline index.js]: It seems that due to the outdated version of Jest, it cannot interpret prefixes like `node:` and thus, tests cannot be executed.

Could you please import without a prefix like this: `const { createHash } = require('crypto');` for now?

I have opened a separate issue #17  regarding the update of dependency packages including Jest, so I will handle it there.

**Review by gladevise — COMMENTED** [review]
I have reviewed it. Please verify only the part where node:crypto is imported.

File: index.js
Comment (Robot-Inventor) [reply index.js]: Oh, I'm sorry. I tested it locally by loading it into another project and forgot to run the test command. 

I fixed it, but it seems to fail cache related tests...If I load it into another project, the cached image is used without any problem, so there must be some cause.

```
 FAIL  tests/test.js
  √ convert bare links to link cards (516 ms)
  √ Inline links are not converted to link cards. (2 ms)
  √ Multiple links in one line are not converted to link cards
  √ Multiple links in one line are not converted to link cards with next-mdx-remote (255 ms)
  √ Decode Url (761 ms)
  × Use cache ogImage (160 ms)
  √ Shorten URL (134 ms)
  √ Use remark-link-card with remark-embedder (6 ms)

  ● Use cache ogImage

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      76 |   const imageElements = parsedOutput.querySelectorAll('img');
      77 |   imageElements.map((element) => {
    > 78 |     expect(element.getAttribute('src').startsWith('/remark-link-card/')).toBe(
         |                                                                          ^
      79 |       true
      80 |     );
      81 |   });

      at map (tests/test.js:78:74)
          at Array.map (<anonymous>)
      at Object.<anonymous> (tests/test.js:77:17)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 7 passed, 8 total
Snapshots:   0 total
Time:        6.052 s
Ran all test suites matching /.\\tests/i.
  console.log
    <a class="rlc-container" href="https://www.npmjs.com/package/remark-link-card">
      <div class="rlc-info">
        <div class="rlc-title">remark-link-card</div>
        <div class="rlc-description">Remark plugin to convert text links to link cards inspired by gatsby-remark-link-card. Latest version: 1.3.1, last published: 3 years ago. Start using remark-link-card in your project by running &#x60;npm i remark-link-card&#x60;. There are no other projects in the npm registry using remark-link-card.</div>
        <div class="rlc-url-container">
          <img class="rlc-favicon" src="https://www.google.com/s2/favicons?domain=www.npmjs.com" alt="remark-link-card favicon" width="16" height="16">
          <span class="rlc-url">www.npmjs.com</span>
        </div>
      </div>
      <div class="rlc-image-container">
          <img class="rlc-image" src="https://static-production.npmjs.com/338e4905a2684ca96e08c7780fc68412.png" alt="remark-link-card" />
        </div>
    </a>

      at Object.<anonymous> (tests/test.js:97:11)
```

Comment (gladevise) [reply index.js]: The reason why the test results differ depending on the environment seems to be that the path for saving cached images is set using `process.cwd()`. As a result, the path where images are downloaded changes based on the directory from which the test command is executed. Consequently, depending on the executed path, there is a possibility that the string for the src attribute may not start with `/remark-link-card/`.

To verify this issue, could you please run the `npm run test` command from the project's root directory?

Comment (Robot-Inventor) [reply index.js]: I tested from root, and I found the cause. This is an OS problem: Windows uses `\` as the path separator, so it does not match `/remark-link-card/` (actually, it would be `\remark-link-card\`) and the test fails. Thus, there is no problem for actual use on Windows, but only the test fails.

Comment (gladevise) [reply index.js]: I see, since I usually develop on Linux, I didn't notice that. Thank you for pointing it out. If only the tests are failing, then I'm thinking I'd like to merge this as is.

Are there any other points of concern?

Comment (Robot-Inventor) [reply index.js]: There are no other concerns.  I think there is no problem with merging.

**Review by gladevise — APPROVED** [review]

**Comment by Suree33** [comment]
I've encountered an issue with a problem with long ogimage filenames in my project, which I believe will be resolved by the changes proposed in this pull request. However, it seems that the deployment is currently blocked on Vercel with an "Authorization required to deploy" error.

Could you please authorize the deployment if everything is ok? It would help me a lot.

**Comment by Suree33** [comment]
> I've encountered an issue with a problem with long ogimage filenames in my project, which I believe will be resolved by the changes proposed in this pull request. However, it seems that the deployment is currently blocked on Vercel with an "Authorization required to deploy" error.
> 
> Could you please authorize the deployment if everything is ok? It would help me a lot.

@gladevise Could you please provide an update on this issue? If there is any additional information or action required from my side, please let me know.

**Comment by Robot-Inventor** [comment]
Hi @Suree33,

Thank you for your interest in this pull request. Perhaps the maintainer is busy. If necessary, you can install the package via GitHub.

```bash
npm install git+https://github.com/gladevise/remark-link-card.git
```

**Pull request merged**
